daphne/src/messages/mod.rs

```diff
@@ -1214,6 +1214,61 @@ pub fn decode_base64url_vec<T: AsRef<[u8]>>(input: T) -> Option<Vec<u8>> {
     URL_SAFE_NO_PAD.decode(input).ok()
 }
 
+fn encode_u16_item_for_version<E: ParameterizedEncode<DapVersion>>(
+    bytes: &mut Vec<u8>,
+    version: DapVersion,
+    item: &E,
+) {
+    match version {
+        DapVersion::Draft07 => {
+            // Cribbed from `decode_u16_items()` from libprio.
+            //
+            // Reserve space for the length prefix.
+            let len_offset = bytes.len();
+            0_u16.encode(bytes);
+
+            item.encode_with_param(&version, bytes);
+            let len = bytes.len() - len_offset - 2;
+            for (offset, byte) in u16::to_be_bytes(len.try_into().unwrap()).iter().enumerate() {
+                bytes[len_offset + offset] = *byte;
+            }
```

Comment:
This can probably be written using [copy_from_slice](https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice)